### PR TITLE
Feature: Create/edit/delete transcript via Video Upload Page

### PR DESCRIPTION
**Pull request by Qubad786**
## [EDUCATOR-1992](https://openedx.atlassian.net/browse/EDUCATOR-1992)
This is a feature branch primarily exposing API utils to create/update/delete transcripts.

### CODE REVIEW

**Comment by Qubad786** [comment]
@mushtaqak @muhammad-ammar have a look.

File: edxval/tests/data/wow.sjson [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+   "start": [10],
+   "end": [100],
+   "text": ["Hi, welcome to edxval."],
+}
```
Comment (mushtaqak) [inline edxval/tests/data/wow.sjson]: nit: New line

Comment (Qubad786) [reply edxval/tests/data/wow.sjson]: this is a test data file -- not code file.

File: edxval/tests/constants.py [hunk 2]
```diff
@@ -373,7 +373,7 @@
 VIDEO_TRANSCRIPT_3PLAY = dict(
     video_id='super-soaker',
     language_code='de',
-    transcript='wow.sjson',
+    transcript='edxval/tests/data/wow.sjson',
```
Comment (muhammad-ammar) [inline edxval/tests/constants.py]: why this change?

File: edxval/tests/test_api.py [hunk 3]
```diff
@@ -1328,7 +1327,7 @@ def test_video_transcript_missing_attribute(self, mock_logger):
             <video_asset>
                 <transcripts>
                     {transcript_xml}
-                    <transcript file_name="wow.sjson" language_code="de" file_format="sjson" provider='3PlayMedia' video_id="{video_id}"/>
+                    <transcript file_name="edxval/tests/data/wow.sjson" language_code="de" file_format="sjson" provider='3PlayMedia' video_id="{video_id}"/>
```
Comment (muhammad-ammar) [inline edxval/tests/test_api.py]: why changes in `file_name` attribute above?

File: edxval/api.py [hunk 4]
```diff
@@ -247,7 +247,7 @@ def get_video_transcript_data(video_ids, language_code):
         try:
             video_transcript = VideoTranscript.objects.get(video_id=video_id, language_code=language_code)
             transcript_data = dict(
-                file_name=video_transcript.transcript.name,
+                file_name=video_transcript.filename,
```
Comment (muhammad-ammar) [inline edxval/api.py]: can this effect existing usages of this `get_video_transcript_data` function?

Comment (Qubad786) [reply edxval/api.py]: No it will not.

File: edxval/tests/constants.py (hunk 2 above)
Comment (Qubad786) [reply edxval/tests/constants.py]: that's the correct file path

File: edxval/tests/test_api.py (hunk 3 above)
Comment (Qubad786) [reply edxval/tests/test_api.py]: As that's changed in data dictionary `VIDEO_TRANSCRIPT_3PLAY`

File: edxval/api.py (hunk 4 above)
Comment (Qubad786) [reply edxval/api.py]: sandbox is up with these changes, you may test there.

File: edxval/tests/data/wow.sjson (hunk 1 above)
Comment (mushtaqak) [reply edxval/tests/data/wow.sjson]: I would still add a new line just for consistency.

File: edxval/models.py [hunk 5]
```diff
@@ -417,6 +417,28 @@ class VideoTranscript(TimeStampedModel):
...
+    @property
+    def filename(self):
+        """
+        Returns readable filename for a transcript
+        """
+        try:
+            video = Video.objects.get(edx_video_id=self.video_id)
+            client_id, __ = os.path.splitext(video.client_video_id)
+            file_name = u'{name}-{language}.{format}'.format(
+                name=client_id,
```
Comment (mushtaqak) [inline edxval/models.py]: Should we reduce file name length ?

File: edxval/models.py [hunk 6]
```diff
@@ -417,6 +417,28 @@ class VideoTranscript(TimeStampedModel):
...
+                language=self.language_code,
+                format=self.file_format
```
Comment (mushtaqak) [inline edxval/models.py]: This may also give us format other than srt, the user may expect srt only ? what do you think?

Comment (Qubad786) [reply edxval/models.py]: this will be the format of the saved (to S3) transcript

File: edxval/models.py (hunk 5 above)
Comment (Qubad786) [reply edxval/models.py]: client_video_id is primarily the file name of an uploaded video, so we don't need to do this.

File: edxval/tests/data/wow.sjson (hunk 1 above)
Comment (Qubad786) [reply edxval/tests/data/wow.sjson]: alright I'll add it
